src/spectral.jl

```diff
@@ -133,3 +133,27 @@ function CombinatorialAdjacency(g::Graph)
     return CombinatorialAdjacency{Float64, typeof(g), typeof(d)}(g,d)
 end
 end # @require
+
+
+"""Returns a sparse node-arc incidence matrix for a directed graph,
+indexed by `[v, e]`. `-1` values indicate `src(e) == v`, while `1`
+values indicates `dst(e) == v`. Otherwise, the value is `0`.
+"""
+function incidence_matrix(g::DiGraph, T::DataType=Int)
+    n_v = nv(g)
+    n_e = ne(g)
+    nz = 2 * ne(g)
+    colpt = collect(1:2:(2*n_e + 1)) # every col has 2 entries
+    rowval = sizehint!(Vector{Int}(), nz)
+    nzval = sizehint!(Vector{T}(), nz)
+    p_one, m_one = one(T), -one(T)
+    for e in edges(g)
```

Comment:
In general, iteration over edges is suboptimal as far as performance is concerned. Is it possible to use the adjacency lists (`fadj` and `badj`) directly?